sn_fault_detection/src/detection.rs

```diff
@@ -832,8 +855,9 @@ mod tests {
             node_count in 4..50, issue_count in 0..50, issue_type in generate_network_startup_msg_issues())
         {
             Runtime::new().unwrap().block_on(async {
-                let nodes = (0..node_count).map(|_| random_xorname()).collect::<Vec<XorName>>();
-                let mut fault_detection = FaultDetection::new(nodes.clone());
+                let nodes = (0..node_count).map(|_| random_xorname()).collect::<BTreeSet<XorName>>();
+                // we're not testing elders vs members here
```

Comment:
Was thinking that this would be consistent with below comment, and so no risk to start wonder if something else is meant here?

```suggestion
                // we're not testing elders vs nodes here
```